src/main/java/com/splunk/hecclient/Indexer.java

```diff
@@ -137,15 +156,61 @@ public boolean send(final EventBatch batch) {
-            resp = httpClient.execute(req, context);
-        } catch (Exception ex) {
-            logBackPressure();
-            log.error("encountered io exception", ex);
-            throw new HecException("encountered exception when post data", ex);
-        }
+        if (hecConfig.kerberosAuthEnabled()) {
+            if (config == null) {
+                config = new Configuration() {
+                    @SuppressWarnings("serial")
+                    @Override
+                    public AppConfigurationEntry[] getAppConfigurationEntry(String name) {
+                        return new AppConfigurationEntry[]{new AppConfigurationEntry("com.sun.security.auth.module.Krb5LoginModule",
+                            AppConfigurationEntry.LoginModuleControlFlag.REQUIRED, new HashMap<String, Object>() {
+                            {
+                                put("useTicketCache", "false");
+                                put("useKeyTab", "true");
+                                put("keyTab", hecConfig.kerberosKeytabLocation());
+                                //Krb5 in GSS API needs to be refreshed so it does not throw the error
+                                //Specified version of key is not available
+                                put("refreshKrb5Config", "true");
+                                put("principal", hecConfig.kerberosPrincipal());
+                                put("storeKey", "false");
+                                put("doNotPrompt", "true");
+                                put("isInitiator", "true");
+                                put("debug", "true");
+                            }
+                        })};
+                    }
+                };
+            }
+            Set<Principal> princ = new HashSet<Principal>(1);
+            princ.add(new KerberosPrincipal(hecConfig.kerberosUser()));
+            Subject sub = new Subject(false, princ, new HashSet<Object>(), new HashSet<Object>());
+            try {
+                LoginContext lc = new LoginContext("", sub, null, config);
+                lc.login();
+                Subject serviceSubject = lc.getSubject();
+                return Subject.doAs(serviceSubject, new PrivilegedAction<HttpResponse>() {
+                    @Override
+                    public HttpResponse run() {
+                        try {
+                            return httpClient.execute(req, context);
+                        } catch (IOException ex) {
+                            throw new ConnectException("Unable to execute HttpClient request", ex);
+                        }
+                    }
+                }).toString();
+            } catch (Exception le) {
+                throw new ConnectException("Unable to authenticate with kerberos server", le);
+            }
+        } else {
+            try {
+                resp = httpClient.execute(req, context);
+            } catch (Exception ex) {
+                logBackPressure();
```

Comment:
This should be called in case of kerberos authorized request failures as well right?